src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -315,50 +316,464 @@ public static void Main()
+    Next (Regular) Block[B2]
+Block[B2] - Exit
+    Predecessors: [B1]
+    Statements (0)
+";
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(src, expectedFlowGraph, DiagnosticDescription.None, parseOptions: TestOptions.RegularPreview);
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_IOperation_FieldsInStructs()
+        {
+            var src = @"
+class Program
+{
+    static void M(A a)
+    /*<bind>*/{
+        _ = a is { Field1.Field2.Field3: null, Field4: null };
+    }/*</bind>*/
+}
+struct A { public B? Field1; public B? Field4; }
+struct B { public C? Field2; }
+struct C { public object Field3; }
+";
+            var expectedDiagnostics = new[]
+            {
+                // (9,22): warning CS0649: Field 'A.Field1' is never assigned to, and will always have its default value
+                // struct A { public B? Field1; public B? Field4; }
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field1").WithArguments("A.Field1", "").WithLocation(9, 22),
+                // (9,40): warning CS0649: Field 'A.Field4' is never assigned to, and will always have its default value
+                // struct A { public B? Field1; public B? Field4; }
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field4").WithArguments("A.Field4", "").WithLocation(9, 40),
+                // (10,22): warning CS0649: Field 'B.Field2' is never assigned to, and will always have its default value
+                // struct B { public C? Field2; }
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field2").WithArguments("B.Field2", "").WithLocation(10, 22),
+                // (11,26): warning CS0649: Field 'C.Field3' is never assigned to, and will always have its default value null
+                // struct C { public object Field3; }
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field3").WithArguments("C.Field3", "null").WithLocation(11, 26)
+            };
+            var comp = CreateCompilation(src, parseOptions: TestOptions.RegularWithExtendedPropertyPatterns);
+            comp.VerifyDiagnostics(expectedDiagnostics);
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var isPattern = tree.GetRoot().DescendantNodes().OfType<IsPatternExpressionSyntax>().Single();
+
+            VerifyOperationTree(comp, model.GetOperation(isPattern), @"
+IIsPatternOperation (OperationKind.IsPattern, Type: System.Boolean) (Syntax: 'a is { Fiel ... ld4: null }')
+  Value:
+    IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: A) (Syntax: 'a')
+  Pattern:
+    IRecursivePatternOperation (OperationKind.RecursivePattern, Type: null) (Syntax: '{ Field1.Fi ... ld4: null }') (InputType: A, NarrowedType: A, DeclaredSymbol: null, MatchedType: A, DeconstructSymbol: null)
+      DeconstructionSubpatterns (0)
+      PropertySubpatterns (2):
```

Comment:
:memo: One way we could solve the problem is representing like the following, using a new IOperation node that has an array of property references onto implicit receivers (as opposed to a chain).

```
          IPropertySubpatternOperation (OperationKind.PropertySubpattern, Type: null) (Syntax: 'Field1.Fiel ... ield3: null')
            Member:
              IExtendedPropertyPatternOperation // new
                Properties:
                      IFieldReferenceOperation: B? A.Field1 (OperationKind.FieldReference, Type: A) (Syntax: 'Field1')
                        Instance Receiver:
                          IInstanceReferenceOperation (ReferenceKind: PatternInput) (OperationKind.InstanceReference, Type: B, IsImplicit) (Syntax: 'Field1')
                      IFieldReferenceOperation: C? B.Field2 (OperationKind.FieldReference, Type: C) (Syntax: 'Field2')
                        Instance Receiver:
                          IInstanceReferenceOperation (ReferenceKind: PatternInput) (OperationKind.InstanceReference, Type: B, IsImplicit) (Syntax: 'Field2')
                      IFieldReferenceOperation: System.Object C.Field3 (OperationKind.FieldReference, Type: System.Object) (Syntax: 'Field3')
                        Instance Receiver:
                          IInstanceReferenceOperation (ReferenceKind: PatternInput) (OperationKind.InstanceReference, Type: C, IsImplicit) (Syntax: 'Field3')
 ```

Reply:
If we decide to use ConvertedType here:

https://github.com/alrz/roslyn/blob/1cb32514462f59221d1e1f09f42a5aa1fa744688/src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs#L2038-L2039

I think it would make sense to also use IConversionOperation to represent a nullable value unwrap in the chain.

Reply:
From chat with Fred and Aleksey, I'll adjust the IOperation tree to take advantage of the fact that the feature is syntactic sugar. `c is { Prop1.Prop2: null } ` -> `c is { Prop1: { Prop2: null } }`. We can do the same for IOperation.